src/sagemaker_containers/_mpi.py

```diff
@@ -0,0 +1,350 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import argparse
+import os
+import shlex
+import socket
+import stat
+import subprocess
+import sys
+import time
+
+import pkg_resources
+from retrying import retry
+
+import sagemaker_containers
+from sagemaker_containers import _errors, _logging, _params, _process, _timeout
+
+logger = _logging.get_logger()
+
+_CHANGE_HOSTNAME_FILE_PATH = pkg_resources.resource_filename(sagemaker_containers.__name__, '/bin/change-hostname.sh')
+
+_SSHD_EXECUTABLE_PATH = '/usr/sbin/sshd'
+
+_SKIP_ENV_VARS_TO_COPY_IN_MPIRUN = ["SM_CURRENT_HOST"]
```

Comment:
does it still work with SM_ADDITIONAL_PARAMS? (that's what didn't work with Chainer)